### PR TITLE
Small features and code clean-up

### PR DESCRIPTION
**Pull request by dcwhite**
See commits for issue numbers

### CODE REVIEW

File: src/Core/Algorithms/BrainStimulator/BiotSavartSolverAlgorithm.cc [hunk 1]
```diff
@@ -56,653 +56,556 @@ ALGORITHM_PARAMETER_DEF(BrainStimulator, VectorBField);
...
+				// result
+				Vector F;
 
-
-	class PieceWiseKernel : public KernelBase
-		{
-			public:
-
-				PieceWiseKernel(const AlgorithmBase* algo, int t ) : KernelBase(algo,t)
+				for (size_t iC0 = 0, iC1 =1, iCV = 0; iC0 < coilNodes_.size(); iC0+=2, iC1+=2, iCV++)
```
Comment (Sailanarmo) [inline src/Core/Algorithms/BrainStimulator/BiotSavartSolverAlgorithm.cc:244]: I'm all for crazy looking for loops but this one seems a little too crazy to me. Is there a better way of making this a little bit more readable?

File: src/Core/Algorithms/BrainStimulator/BiotSavartSolverAlgorithm.cc [hunk 2]
```diff
@@ -740,94 +642,72 @@ bool BiotSavartSolverAlgorithm::run(FieldHandle mesh, FieldHandle coil, MatrixHa
...
-  output[Parameters::VectorBField] = outdata1;
+	auto oports = get(Parameters::OutType).toInt();
+	std::map<int, DenseMatrixHandle> cases;
+	if (oports == 3)
+		cases = {{1, nullptr}, {2, nullptr}};
+	else
+		cases[oports] = nullptr;
 
-  if(!run(mesh, coil, outdata2, 2))
+  for (auto& c : cases)
```
Comment (Sailanarmo) [inline src/Core/Algorithms/BrainStimulator/BiotSavartSolverAlgorithm.cc:702]: I guess this is pedantic of me, but is there a reason c shouldn't be constant?

File: src/Interface/Application/StateViewer.cc [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+#include <QMenu>
+#include <Interface/Application/StateViewer.h>
+#include <Interface/Application/NetworkEditor.h>
+#include <Interface/Application/ModuleProxyWidget.h>
+#include <Interface/Application/ModuleWidget.h>
+
+using namespace SCIRun::Gui;
+using namespace SCIRun::Core;
+
+static const std::map<int, QString> variableTypeLabels =
```
Comment (Sailanarmo) [inline src/Interface/Application/StateViewer.cc:39]: I guess my only question is why this doesn't belong privately to the class? Will a future class or something have access to this in the future?

File: src/Core/Algorithms/BrainStimulator/BiotSavartSolverAlgorithm.cc (hunk 1 above)
Comment (dcwhite) [reply src/Core/Algorithms/BrainStimulator/BiotSavartSolverAlgorithm.cc:244]: This was old code, it just got a white-space fix. I'm not touching the algorithms.

File: src/Core/Algorithms/BrainStimulator/BiotSavartSolverAlgorithm.cc (hunk 2 above)
Comment (dcwhite) [reply src/Core/Algorithms/BrainStimulator/BiotSavartSolverAlgorithm.cc:702]:
```suggestion
  for (const auto& c : cases)
```

File: src/Core/Algorithms/BrainStimulator/BiotSavartSolverAlgorithm.cc (hunk 1 above)
Comment (Sailanarmo) [reply src/Core/Algorithms/BrainStimulator/BiotSavartSolverAlgorithm.cc:244]: oh fair enough. If you didn't write it, no touch. Cool.

File: src/Core/Algorithms/BrainStimulator/BiotSavartSolverAlgorithm.cc (hunk 2 above)
Comment (dcwhite) [reply src/Core/Algorithms/BrainStimulator/BiotSavartSolverAlgorithm.cc]: Done!

File: src/Interface/Application/StateViewer.cc (hunk 3 above)
Comment (dcwhite) [reply src/Interface/Application/StateViewer.cc:39]: Probably not. But it's already static/private to the translation unit. A private member would be declared in the header and expose it even more.

Comment (Sailanarmo) [reply src/Interface/Application/StateViewer.cc:39]: fair enough. Alright, I will approve!

**Comment by dcwhite** [comment]
Oops the github.com-added code fix actually doesn't compile. I'll need to revert it and use non-const `auto&`

**Comment by dcwhite** [comment]
@allywarner Sorry could you click the approve button on this one again? I added one tiny thing

**Pull request merged**